Core/PersistentPixel.swift

```diff
@@ -0,0 +1,299 @@
+            }
+        }
+    }
+
+    public func fireDailyAndCount(pixel: Pixel.Event,
+                                  error: Swift.Error? = nil,
+                                  withAdditionalParameters additionalParameters: [String: String],
+                                  includedParameters: [Pixel.QueryParameters] = [.appVersion],
+                                  completion: @escaping ([Error]) -> Void = { _ in }) {
+        let dispatchGroup = DispatchGroup()
+
+        dispatchGroup.enter() // onDailyComplete
+        dispatchGroup.enter() // onCountComplete
+
+        var dailyPixelStorageError: Error?
+        var countPixelStorageError: Error?
+
+        let fireDate = dateGenerator()
+        let dateString = dateFormatter.string(from: fireDate)
+        var additionalParameters = additionalParameters
+        additionalParameters[PixelParameters.originalPixelTimestamp] = dateString
+
+        Logger.general.debug("Firing persistent daily/count pixel named \(pixel.name)")
+
+        dailyPixelFiring.fireDailyAndCount(
+            pixel: pixel,
+            error: error,
+            withAdditionalParameters: additionalParameters,
+            includedParameters: includedParameters,
+            pixelFiring: Pixel.self,
+            dailyPixelStore: DailyPixel.storage,
+            onDailyComplete: { dailyError in
+                if dailyError != nil {
+                    do {
+                        if let error { additionalParameters.appendErrorPixelParams(error: error) }
+                        Logger.general.debug("Saving persistent daily pixel named \(pixel.name)")
+                        try self.persistentPixelStorage.append(pixels: [
+                            PersistentPixelMetadata(
+                                eventName: pixel.name,
+                                pixelType: .daily,
+                                additionalParameters: additionalParameters,
+                                includedParameters: includedParameters
+                            )
+                        ])
+                    } catch {
+                        dailyPixelStorageError = error
+                    }
+                }
+
+                dispatchGroup.leave()
+            }, onCountComplete: { countError in
+                if countError != nil {
+                    do {
+                        if let error { additionalParameters.appendErrorPixelParams(error: error) }
+                        Logger.general.debug("Saving persistent count pixel named \(pixel.name)")
+                        try self.persistentPixelStorage.append(pixels: [
+                            PersistentPixelMetadata(
+                                eventName: pixel.name,
+                                pixelType: .count,
+                                additionalParameters: additionalParameters,
+                                includedParameters: includedParameters
+                            )
+                        ])
```

Comment:
Validating assumptions here: typically daily/count combo is used for pixels that may fire more frequently and we want to understand affected devices - in case 'count' pixel floods us with data it can easily overflow the array (discarding other pixels) so we only get one signal.

I'm not saying we need to handle this now, but is seems like a hole in the current approach.

Reply:
I've left this alone for the time being as I feel I'll still get some value from the current state, but I'd like to come back to it next week. At the same time, I'd also like to triage and add a storage error pixel, per a previous comment. Let me know if you feel strongly about addressing this one now before it's merged though.